client/components/domains/search-filters/tld-filter-bar.jsx

```diff
@@ -0,0 +1,206 @@
+		);
+
+		const tldSet = new Set( [ ...this.props.filters.tlds, newTld ] );
+		if ( isCurrentlySelected ) {
+			tldSet.delete( newTld );
+		}
+
+		const tlds = [ ...tldSet ];
+		this.props.onChange( { tlds }, { shouldSubmit: true } );
+	};
+	handleFiltersReset = () => {
+		this.togglePopover();
+		this.props.onReset( 'tlds' );
+	};
+	handleFiltersSubmit = () => {
+		this.togglePopover();
+		this.props.onSubmit();
+	};
+	handleTokenChange = newTlds => {
+		const tlds = [
+			...new Set( [
+				...this.getSelectedSuggestedTlds(),
+				...newTlds.filter( tld => includes( this.props.availableTlds, tld ) ),
+			] ),
+		];
+
+		this.props.onChange( { tlds } );
+	};
+
+	togglePopover = () => {
+		this.setState( {
+			showPopover: ! this.state.showPopover,
+		} );
+	};
+
+	getAvailableSuggestedTlds() {
+		return this.props.availableTlds.slice( 0, this.props.numberOfTldsShown );
+	}
+
+	getAvailableRegularTlds() {
+		return this.props.availableTlds.slice( this.props.numberOfTldsShown );
+	}
+
+	getSelectedSuggestedTlds() {
+		return this.props.filters.tlds.filter( tld =>
+			includes( this.getAvailableSuggestedTlds(), tld )
+		);
+	}
+
+	getSelectedRegularTlds() {
+		return this.props.filters.tlds.filter(
+			tld => ! includes( this.getAvailableSuggestedTlds(), tld )
+		);
+	}
+
+	render() {
+		const isKrackenUi = config.isEnabled( 'domains/kracken-ui/tld-filters' );
+		if ( ! isKrackenUi ) {
+			return null;
+		}
+
+		if ( this.props.showPlaceholder ) {
+			return this.renderPlaceholder();
+		}
+
+		const { filters: { tlds }, lastFilters: { tlds: selectedTlds }, translate } = this.props;
+		const hasFilterValue = tlds.length > 0;
+
+		return (
+			<CompactCard className="search-filters__buttons">
+				{ this.getAvailableSuggestedTlds().map( ( tld, index ) => (
+					<Button
+						className={ classNames( { 'is-active': includes( selectedTlds, tld ) } ) }
+						data-selected={ includes( selectedTlds, tld ) }
+						data-index={ index }
+						key={ tld }
+						onClick={ this.handleButtonClick }
+						value={ tld }
+					>
+						.{ tld }
+					</Button>
+				) ) }
+				<Button
+					className={ classNames( { 'is-active': hasFilterValue } ) }
+					onClick={ this.togglePopover }
+					ref={ this.bindButton }
+					key="popover-button"
+				>
+					{ translate( 'More Extensions', {
+						context: 'TLD filter button',
+						comment: 'Refers to top level domain name extension, such as ".com"',
+					} ) }
+					<Gridicon icon="chevron-down" size={ 24 } />
+				</Button>
+
+				{ this.state.showPopover && this.renderPopover() }
+			</CompactCard>
+		);
+	}
+
+	renderPopover() {
+		const { translate } = this.props;
+
+		return (
+			<Popover
+				autoPosition={ false }
+				className="search-filters__popover"
+				context={ this.button }
+				isVisible={ this.state.showPopover }
+				onClose={ this.togglePopover }
+				position="bottom left"
+			>
+				<FormFieldset className="search-filters__token-field-fieldset">
+					<TokenField
```

Comment:
Seems this needs a `maxSuggestions={ 500 }` or something, otherwise it defaults to `100`, which is probably not what we need/want (since it cuts of the TLD list quite early).

Reply:
👍